### PR TITLE
Allow colon as valid host pattern char to support ipv6 entries in kno…

### PR DESCRIPTION
**Pull request by FliegenKLATSCH**
…wn_hosts (SSHD-745)

### CODE REVIEW

**Comment by lgoldstein** [comment]
Please add some unit tests and I will merge it...

**Comment by lgoldstein** [comment]
Please resolve the conflicts and let me know when you feel it is ready to merge

**Comment by FliegenKLATSCH** [comment]
Should be fine now.
Do you already have any plannings regarding the next release (date)?

**Comment by lgoldstein** [comment]
>> Should be fine now.

Thx - will check it out and merge it in the next few days

>> Do you already have any plannings regarding the next release (date)?

Not really - there are still some features/bugs/improvements that we consider. Tentatively I would say early June 2020....

File: sshd-core/src/test/resources/org/apache/sshd/client/keyverifier/known_hosts [hunk 1]
```diff
@@ -13,3 +13,5 @@ server.sshd.apache.org,10.23.222.240 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbml
 
 # non-standard port overrides
 [issues.apache.org]:5637 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBCWDHD00Ltb5fmmL9cFLTqWqxgJHwsxbiZgL632CXqbDmf69wA+8GSP96rtIix2d5aGXyh/kXMbSMjPgIx+n7p0=
+
+fe80::7780:db3:a57:6a9 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBCWDHD00Ltb5fmmL9cFLTqWqxgJHwsxbiZgL632CXqbDmf69wA+8GSP96rtIix2d5aGXyh/kXMbSMjPgIx+n7p0=
```
Comment (lgoldstein) [inline sshd-core/src/test/resources/org/apache/sshd/client/keyverifier/known_hosts:17]: Note that it does not cover [link-local](https://en.wikipedia.org/wiki/IPv6#Link-local_address) address, or [scoped address](https://en.wikipedia.org/wiki/IPv6_address) - but for now, it should be good enough

**Comment by lgoldstein** [comment]
Merged in commit [edb7b3fe4b0b1d67d42559a9b75cd8c95f87e814](https://github.com/apache/mina-sshd/commit/edb7b3fe4b0b1d67d42559a9b75cd8c95f87e814) please close the PR

**Pull request closed without merging**